taskmanager/src/Adapters/Incoming/JSON/AddTask.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Adapters\Incoming\JSON;
+
+use App\Domain\Ports\DTO\NewTask;
+use App\Domain\Process\TaskService;
+
+class AddTask implements \App\Domain\Ports\Incoming\AddTask
+{
+    /** @var TaskService */
+    private $taskService;
+
+    /**
+     * AddTask constructor.
+     * @param $taskService
+     */
+    public function __construct(TaskService $taskService)
+    {
+        $this->taskService = $taskService;
+    }
+
+    public function addTask($data)
+    {
+        $data = json_decode($data, true);
+
+        $task = new NewTask();
```

Comment:
i dislike it, you should pass all params as arguments of the class

Reply:
+1